packages/sanity/src/desk/components/pane/PaneLayout.tsx

```diff
@@ -50,6 +51,16 @@ export function PaneLayout(
     if (!collapsed && onExpand) onExpand()
   }, [collapsed, onCollapse, onExpand])
 
+  // Force popover position update when panes change
+  // This is needed to ensure that any open popovers is positioned correctly if a pane is toggled
+  // between collapsed and expanded. We normally use the boundary element, but that won't work
+  // inside the PTE
+  useEffect(() => {
```

Comment:
We could do something like this to prevent the update function from running when there are no popovers in the DOM:

```ts
function hasPopovers() {
  const attr = '[data-ui="Popover"]'
  const popovers = document.querySelectorAll(attr)
  return popovers.length > 0
}
```

```ts
  useEffect(() => {
    if (state.panes.length && hasPopovers()) {
      forcePopoverUpdate?.()
    }
  }, [state.panes, forcePopoverUpdate])

```

Reply:
I agree that this might be a good idea, but is there precedence in the studio of querying for HTML attributes like this?